src/ansys/mapdl/core/hpc/pyhps.py

```diff
@@ -73,487 +73,846 @@ def get_value_from_json_or_default(
+
+    def _validate_inputs(self, inputs):
+        """Validate inputs inputs"""
+
+        if self.mode != "python":
+            raise ValueError("Inputs are not supported when using APDL or shell files.")
+
+        if inputs is None:
+            inputs = []
+
+        if isinstance(inputs, str):
+            inputs = inputs.split(",")
+
+        return inputs
+
+    def _validate_outputs(self, outputs):
+        """Validate outputs inputs"""
+
+        if self.mode != "python":
+            raise warn(
+                f"""Outputs are not directly supported when using APDL or shell files.
+                However you can still write the parameters you want to the
+                file {self._output_parms_file} using the python format
+                ('PARAMETER=VALUE')."""
             )
+
+        if outputs is None:
+            outputs = []
+
+        if isinstance(outputs, str):
+            outputs = outputs.split(",")
+
+        return outputs
+
+    def _validate_main_file(self, main_file):
+        """Validate main_file input"""
+
+        if not os.path.exists(main_file):
+            raise ValueError(f"The file {main_file} must exist.")
+
+        logger.debug(f"Main file is in: {main_file}.")
+        return main_file
+
+    def _validate_shell_file(self, shell_file):
+        """Validate shell_file input"""
+        return shell_file
+
+    def _validate_extra_files(self, extra_files):
+        """Validate extra_files input"""
+        if isinstance(extra_files, str):
+            extra_files = extra_files.split(",")
+        elif extra_files is None:
+            extra_files = []
+
+        if extra_files and not all([os.path.exists(each) for each in extra_files]):
+            raise ValueError("One or more extra files does not exist.")
+
+        return extra_files
+
+    def _validate_output_files(self, output_files):
+        """Validate output_files input"""
+        if not output_files:
+            output_files = []
+        elif isinstance(output_files, str):
+            output_files = output_files.split(",")
+
+        return output_files
+
+    def _validate_requirements_file(self, requirements_file):
+        """Validate requirements_file input"""
+        return requirements_file
+
+    def _validate_python(self, python):
+        """Validate python inputs"""
+        if python not in [2, 2.7, 3, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12]:
+            warn(f"Version of Python {python} might not be supported by the cluster.")
+        return python
+
+    def _validate_num_cores(self, num_cores: int):
+        """Validate num_cores inputs"""
+        if not num_cores:
+            num_cores = 0
+
+        return int(num_cores)
+
+    def _validate_memory(self, memory: int):
+        """Validate memory inputs"""
+        if not memory:
+            memory = 0
+
+        return int(memory)
+
+    def _validate_disk_space(self, disk_space: float):
+        """Validate disk_space inputs"""
```

Comment:
```suggestion
        """Validate disk space."""
```
Maybe "Get the disk space available."?